### PR TITLE
RE2022-241: Microtrait update

### PR DESCRIPTION
**Pull request by dakotablair**
- [x] [RE2022-241](https://kbase-jira.atlassian.net/browse/RE2022-241): HeatMap component refactor
  - [x] tooltips
    - [x] cursor following tooltip which locks on click
    - [x] content selectable for clipboard management
  - [x] constrain axis ranges to data
    - [x] If a user navigates outside the range of the data, the view should revert to showing the data.
  - [x] labels render vertically
  - [x] some things we get for free with plotly.js:
    - [x] quick rendering of ~30k cells
    - [x] zoom, scroll
    - [x] labels adjust to available space
    - [x] column labels at top
    - [x] update width on window resize

[RE2022-241]: https://kbase-jira.atlassian.net/browse/RE2022-241?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/features/collections/data_products/Microtrait.tsx [hunk 1]
```diff
@@ -143,8 +149,8 @@ const useMicrotrait = (collection_id: string | undefined) => {
   );
 
   // HeatMap cell query
-  const microtraitQuery = getMicroTrait.useQuery(heatMapParams, {
-    skip: !collection_id,
+  fonst microtraitQuery = getMicroTrait.useQuery(heatMapParams, {
+    fkip: !collection_id,
```
Comment (dakotablair) [inline src/features/collections/data_products/Microtrait.tsx:153]: Why did I do this? Also how? I would love to know how I managed this odd change.

File: src/features/collections/data_products/HeatMap.module.scss [hunk 2]
```diff
@@ -2,10 +2,76 @@
 
 $label-len: 20em;
 
+:global(div.dragcover) {
```
Comment (dauglyon) [inline src/features/collections/data_products/HeatMap.module.scss:5]: any way to get more specific here? I worry it may affect other plots in the future

File: src/features/navigator/NarrativeControl/NarrativeControl.test.tsx [hunk 3]
```diff
@@ -14,6 +14,10 @@ import {
 } from './NarrativeControl.stories';
 
 describe('The <NarrativeControl /> component...', () => {
+  beforeAll(() => {
+    window.gtag = jest.fn();
```
Comment (dauglyon) [inline src/features/navigator/NarrativeControl/NarrativeControl.test.tsx:18]: "

File: src/features/navigator/NarrativeView.test.tsx [hunk 4]
```diff
@@ -48,6 +48,7 @@ consoleWarn.mockImplementation(noOp);
 describe('The <NarrativeView /> component...', () => {
   beforeAll(() => {
     enableFetchMocks();
+    window.gtag = jest.fn();
```
Comment (dauglyon) [inline src/features/navigator/NarrativeView.test.tsx:51]: "

File: src/features/navigator/SortSelect.test.tsx [hunk 5]
```diff
@@ -5,34 +5,40 @@ import { MemoryRouter as Router } from 'react-router-dom';
...
-        <SortSelect sort={'lex'} />
-      </Router>
-    </Provider>
-  );
-  expect(container).toBeTruthy();
-  expect(container.querySelector('.react-select.sort')).toBeInTheDocument();
-});
+describe('The <SortSelect /> component...', () => {
+  beforeAll(() => {
+    window.gtag = jest.fn();
```
Comment (dauglyon) [inline src/features/navigator/SortSelect.test.tsx:10]: "

File: src/features/profile/profileSlice.test.tsx [hunk 6]
```diff
@@ -9,6 +9,10 @@ import { useLoggedInProfileUser } from './profileSlice';
 
 let testStore = createTestStore({});
 describe('useLoggedInProfileUser', () => {
+  beforeAll(() => {
+    window.gtag = jest.fn();
```
Comment (dauglyon) [inline src/features/profile/profileSlice.test.tsx:13]: "... etc

File: src/index.tsx [hunk 7]
```diff
@@ -6,6 +6,8 @@ import './index.scss';
 import App from './app/App';
 import { store } from './app/store';
 import reportWebVitals from './reportWebVitals';
+import { ThemeProvider } from '@mui/material';
+import { theme } from './theme';
```
Comment (dauglyon) [inline src/index.tsx:10]: I'm not sure why this change is in this PR

File: src/features/collections/MatchModal.tsx [hunk 8]
```diff
@@ -314,6 +314,10 @@ const CreateMatch = ({ collectionId }: { collectionId: string }) => {
   const idNarrative = useId();
   const idDataObject = useId();
 
+  // React complains that the provided ref is useless for a FC.
+  const registerMatcher = Object.fromEntries(
+    Object.entries(register('matcher')).filter(([key]) => key !== 'ref')
+  );
```
[the remark this answers is not in the document]
Comment (dakotablair) [reply src/features/collections/MatchModal.tsx:320]: See the comment, React suggests that functional components can't make use of refs this way in any case. I was convinced that we had done this elsewhere before also, but I kept looking for the code and could not find it.

File: src/features/collections/data_products/HeatMap.module.scss (hunk 2 above)
Comment (dakotablair) [reply src/features/collections/data_products/HeatMap.module.scss:9]: Unfortunately, this is plotly behavior which is not very configurable. As we discussed, the ways to improve this situation are complicated and best saved for a circumstance where a problem is encountered.

File: src/index.tsx (hunk 7 above)
Comment (dakotablair) [reply src/index.tsx]: The original branch was rebased from the demo branch before these changes were merged into `main`, so now that they are another rebase fixes this issue.

**Review by dauglyon — APPROVED** [review]
